### PR TITLE
Add main README for jdt.debug and clean-up CONTRIBUTING file

### PR DESCRIPTION
**Pull request by HannesWell**
## What it does
Add main README for jdt.debug and clean-up CONTRIBUTING file.
The content of both is based on the content of the corresponding files in jdt.core/ui.

## Author checklist

- [ ] I have thoroughly tested my changes
- [x] The change is following the [coding conventions](https://wiki.eclipse.org/Platform/How_to_Contribute#Coding_Conventions)
- [x] I have signed the [Eclipse Contributor Agreement (ECA)](https://www.eclipse.org/legal/ECA.php)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+JDT Debug implements Java debugging support and works with any JDPA-compliant target Java VM.
+It is implemented on top of the language independent "debug model" provided by the platform debugger. 
+JDT debug provides the following debugging features:
+
+- Launching of a Java VM in either run or debug mode
+- Attaching to a running Java VM
+- Expression evaluation in the context of a stack frame
+- Scrapbook pages for interactive Java code snippet evaluation
+- Dynamic class reloading where supported by Java virtual machine
```
Comment (HannesWell) [inline README.md:11]: This section is mainly based on the content of
https://eclipse.dev/eclipse/debug/index.php

File: CONTRIBUTING [hunk 2]
```diff
@@ -1,59 +1,3 @@
...
-For more information, please see the Eclipse Committer Handbook:
-https://www.eclipse.org/projects/handbook/#resources-commit
-
-## Contact
-
-Contact the project developers via the project's "dev" list.
-
-* https://dev.eclipse.org/mailman/listinfo/jdt-dev
\ No newline at end of file
+See https://github.com/eclipse-jdt/.github/blob/main/CONTRIBUTING.md
```
Comment (jukzi) [inline CONTRIBUTING:3]: is that file needed at all? when its deleted it shout be automatically taken from .github

File: README.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+For more information about JDT Debug refer to the [JDT Debug website](https://eclipse.dev/eclipse/debug/index.php).
+For general information about JDT and important links, refer to the [JDT wiki page](https://github.com/eclipse-jdt/eclipse.jdt.core/wiki) or the [JDT project overview page](https://projects.eclipse.org/projects/eclipse.jdt).
+
+## Contributing
```
Comment (jukzi) [inline README.md:16]: does this file need a Contributing section? The existing of Contributing.md should be self explaining

**Comment by jukzi** [comment]
@SougandhS can you review, please?

**Comment by SougandhS** [comment]
> @SougandhS can you review, please?


Sure @jukzi

File: README.md (hunk 3 above)
Comment (SougandhS) [reply README.md:16]: I agree with you on this. looks good without Contributing section 
image

File: CONTRIBUTING (hunk 2 above)
Comment (SougandhS) [reply CONTRIBUTING:3]: Removal will be a good choice.

Comment (HannesWell) [reply CONTRIBUTING]: > is that file needed at all? when its deleted it shout be automatically taken from .github

I cannot say if there are legal requirements or alike and I also don't know if it's inherited. I just know that many, if not all, repos in the platform, equinox, PDE organization have it. Maybe just to avoid dead links?

But I removed it now.

File: README.md (hunk 3 above)
Comment (HannesWell) [reply README.md:16]: > does this file need a Contributing section? The existing of Contributing.md should be self explaining

The `contributing.md` is now removed and even if it's inherited from the organization I think it's good to have a few words about contributions and to provide the link to the orga's contributing section explicitly.

Furthermore I wanted to add the Oomph setup button for the setup to be added for repository (see https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/issues/2430) in that section in a follow-up PR, similar to how it is e.g. in the platform repository:
https://github.com/eclipse-platform/eclipse.platform.ui/?tab=readme-ov-file#how-to-contribute

Therefore I would be in favor to keep this.

File: CONTRIBUTING (hunk 2 above)
Comment (jukzi) [reply CONTRIBUTING]: It was a honest question. I know from https://docs.github.com/de/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors that https://github.com/eclipse-jdt/eclipse.jdt.ui/contribute does link it. But i do not know who would know how to go to that URL, i don't see a link to it anywhere and have never used it myself

**Comment by jukzi** [comment]
lets try

**Comment by HannesWell** [comment]
Thanks for submitting.

> https://github.com/eclipse-jdt/eclipse.jdt.debug/contribute now shows the .github link :-)

That's nice, I didn't know this was possible.
But I don't know either, where this link is accessible from the UI?
But if it works well in general, we could delete many of the contributing files in many repositories that are more or less empty.

**Comment by jukzi** [comment]
> we could delete many of the contributing files

yes, please. While Readme makes sense per repository, the governing rules belong to the organization and are same for all repos the org owns.

**Comment by HannesWell** [comment]
> > we could delete many of the contributing files
> 
> yes, please. While Readme makes sense per repository, the governing rules belong to the organization and are same for all repos the org owns.

Created https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/issues/2448 to track that.
